12-offer-encoding.md

```diff
@@ -551,144 +585,100 @@ using the `onion_message` `invoice` field.
-    - otherwise:
-      - MUST NOT set `refund_signature`
+      - MUST NOT set `invoice_request_quantity`
+    - MUST set `invoice_request_payer_key` to `offer_node_id`.
+    - MUST set `invoice_request_payer_info`
+	  - SHOULD set it to at least 16 random bytes.
 
 A reader of an invoice:
-  - MUST reject the invoice if `signature` is not a valid signature using `node_id` as described in [Signature Calculation](#signature-calculation).
-  - MUST reject the invoice if `amount` is not present.
-  - MUST reject the invoice if `description` is not present.
-  - MUST reject the invoice if `created_at` is not present.
-  - MUST reject the invoice if `payment_hash` is not present.
-  - if `relative_expiry` is present:
-    - MUST reject the invoice if the current time since 1970-01-01 UTC is greater than `created_at` plus `seconds_from_creation`.
+  - MUST reject the invoice if `signature` is not a valid signature using `offer_node_id` as described in [Signature Calculation](#signature-calculation).
+  - MUST reject the invoice if `invoice_request_amount` is not present.
+  - MUST reject the invoice if `offer_description` is not present.
+  - MUST reject the invoice if `invoice_created_at` is not present.
+  - MUST reject the invoice if `invoice_payment_hash` is not present.
+  - if `invoice_features` contains unknown _odd_ bits that are non-zero:
+    - MUST ignore the bit.
+  - if `invoice_features` contains unknown _even_ bits that are non-zero:
+    - MUST reject the invoice.
+  - if `invoice_relative_expiry` is present:
+    - MUST reject the invoice if the current time since 1970-01-01 UTC is greater than `invoice_created_at` plus `seconds_from_creation`.
   - otherwise:
-    - MUST reject the invoice if the current time since 1970-01-01 UTC is greater than `created_at` plus 7200.
-  - MUST reject the invoice if `paths` is not present or is empty.
-  - MUST reject the invoice if `blindedpay` is not present.
-    - MUST reject the invoice if `blindedpay` does not contain exactly one `blinded_payinfo` per `blinded_path`.
-  - if `blinded_capacities` is present:
-    - MUST reject the invoice if `blinded_capacities` does not contain exactly one `u64` per `blinded_path`.
-  - SHOULD confirm authorization if `msat` is not within the amount range authorized.
+    - MUST reject the invoice if the current time since 1970-01-01 UTC is greater than `invoice_created_at` plus 7200.
+  - MUST reject the invoice if `invoice_paths` is not present or is empty.
+  - MUST reject the invoice if `invoice_blindedpay` is not present.
+  - MUST reject the invoice if `invoice_blindedpay` does not contain exactly one `blinded_payinfo` per `invoice_paths`.`blinded_path`.
+- MUST reject the invoice if `features` in any `blinded_payinfo` has any unknown even bits set.
+  - SHOULD confirm authorization if `invoice_request_amount`.`msat` is not within the amount range authorized.
   - if the invoice is a reply to an `invoice_request`:
```

Comment:
I think this has an extra indentation and below?

Reply:
Actually, indentation is wrong on  651 "- MUST reject the invoice if features"... fixed.